### ISSUE TITLE
Graphical Lag with the mod

### ISSUE THREAD

**Issue by Saereth** [MrRiegel/Storage-Network#24]
latest version of the mod as of yesterday. We have a moderate setup, probably a couple hundred wires and half a dozen storages linked up but lately we've been getting some bad graphical lag. F3 showing hundreds of updates, after much trouble shooting we finally removed the cable master and suddenly all the lag was gone. So it would appear that once you start getting into a decent sized setup you run into pretty extreme fps loss. Was hitting as low as 10fps at times on a high end gaming pc. 

Optifine seemed to help immensely once I added that in for what it's worth.

**Comment by KidsDontPlay** [comment]
can you give me the world save?

**Comment by Saereth** [comment]
It's on a sever I'll talk to the owner about it this evening

**Comment by MrGraphic** [comment]
So I tested this on my side with some "very" complicated network in both 1.8 and 1.8.1
Did not tried smp but I doubt this will make a difference in this case.
Of course fps drop is expected with that amount of connected TileEntities - don't know if @MrRiegel is using ISBMs or FastTESRs or other special case renderers but it is expected. In my case FPS drops from 120 to around 40-45 and my processor isn't that high end anymore.
I did not notice any difference between those two version 
There was also no difference if network had cable master or not, same for moving items or not.
There was short lag on first connection with CM. Other than that 
Stats;
- 1200 grey network cables
- 240 storage cables connected to
- the same amount of chests
- 90 import cables connected to
- 90 chests that are connected to another
- 90 storage ones that are connected to
  -3 super fast import cables sucking cobble in order to saturate whole network.
  The power usage is way higher than CM can handle hehe.
  Pretty picture of my setup:
  ![2016-03-18_22 00 15](https://cloud.githubusercontent.com/assets/4780141/13893856/eb682396-ed59-11e5-9ba3-b200ad22b018.png)
  ![2016-03-18_21 59 08](https://cloud.githubusercontent.com/assets/4780141/13893861/fd81d22a-ed59-11e5-8fa1-e696f7a6d1c7.png)

**Comment by MrGraphic** [comment]
BTW I do not recommend this kind of networks other than a test env as apparently it is too much data for MC Anvil engine;

``` java
[File IO Thread/ERROR]: Failed to save chunk
java.io.UTFDataFormatException: encoded string too long: 98113 bytes
    at java.io.DataOutputStream.writeUTF(Unknown Source) ~[?:1.8.0_74]
    at java.io.DataOutputStream.writeUTF(Unknown Source) ~[?:1.8.0_74]
    at net.minecraft.nbt.NBTTagString.func_74734_a(NBTTagString.java:28) ~[ea.class:?]
    at net.minecraft.nbt.NBTTagCompound.func_150298_a(NBTTagCompound.java:392) ~[dn.class:?]
    at net.minecraft.nbt.NBTTagCompound.func_74734_a(NBTTagCompound.java:24) ~[dn.class:?]
    at net.minecraft.nbt.NBTTagList.func_74734_a(SourceFile:45) ~[du.class:?]
    at net.minecraft.nbt.NBTTagCompound.func_150298_a(NBTTagCompound.java:392) ~[dn.class:?]
    at net.minecraft.nbt.NBTTagCompound.func_74734_a(NBTTagCompound.java:24) ~[dn.class:?]
    at net.minecraft.nbt.NBTTagCompound.func_150298_a(NBTTagCompound.java:392) ~[dn.class:?]
    at net.minecraft.nbt.NBTTagCompound.func_74734_a(NBTTagCompound.java:24) ~[dn.class:?]
    at net.minecraft.nbt.CompressedStreamTools.func_150663_a(CompressedStreamTools.java:111) ~[dx.class:?]
    at net.minecraft.nbt.CompressedStreamTools.func_74800_a(CompressedStreamTools.java:101) ~[dx.class:?]
    at net.minecraft.world.chunk.storage.AnvilChunkLoader.func_183013_b(AnvilChunkLoader.java:232) ~[anj.class:?]
    at net.minecraft.world.chunk.storage.AnvilChunkLoader.func_75814_c(AnvilChunkLoader.java:210) [anj.class:?]
    at net.minecraft.world.storage.ThreadedFileIOBase.func_75736_b(SourceFile:37) [auc.class:?]
    at net.minecraft.world.storage.ThreadedFileIOBase.run(SourceFile:30) [auc.class:?]
    at java.lang.Thread.run(Unknown Source) [?:1.8.0_74]
```

**Comment by KidsDontPlay** [comment]
I don't think it's the fault of the cables.
@Saereth wrote, that the problem was gone by breaking the cable master.

**Comment by Saereth** [comment]
yes, it always wasnt anywhere near that many, 300 cables tops, 3 request tables, 1 master. They were connected to storage drawer controllers (2) and 3 modular storage systems from rftools if that makes any difference. I havent heard from the server owner about a world save yet.

**Comment by MrGraphic** [comment]
@MrRiegel, saving chunk failure ? Pretty sure mc is unable to save chunk with cable master that is connected to that big network ^^  - ripping part of this monster helps :).

Ok, I hooked a bunch of rf modular storage blocks filled with stuff plus a bunch of storage drawers controllers with few drawers each - that did not change much in frames.
The only thing that causes more drops ( because of chunk redraws ) is Indicator block if I hook it additionally and it is not constant drop, more like every half a second. Plus in this case it cost me like 4ms of TPS.
It is worth to mention that I did not set anykind of filters in cables.
